client/components/command-palette/use-command-palette.tsx

```diff
@@ -88,13 +88,53 @@ export const useCommandPalette = ( {
 	const { currentSiteId } = useCurrentSiteRankTop();
 
 	// Call the generateCommandsArray function to get the commands array
-	let commands = useCommandsArrayWpcom( { setSelectedCommandName } );
+	const commands = useCommandsArrayWpcom( {
+		setSelectedCommandName,
+	} );
 
-	if ( 'function' === typeof filter ) {
-		commands = commands.filter( filter );
-	}
+	const currentPath = useSelector( ( state: object ) => getCurrentRoute( state ) );
+
+	// Filter out commands that have context
+	const commandHasContext = ( paths: string[] = [] ): boolean =>
+		paths.some( ( path ) => currentPath.includes( path ) ) ?? false;
+
+	// Ensure that View My Sites command ranks the highest
+	const viewMySitesSort = ( a: { name: string }, b: { name: string } ) => {
+		const isViewMySitesWithContextual = a.name === 'viewMySites';
+		const isViewMySitesNoContextual = b.name === 'viewMySites';
+
+		if ( isViewMySitesWithContextual && ! isViewMySitesNoContextual ) {
+			return -1; // "viewMySites" comes first over contextual commands
+		} else if ( ! isViewMySitesWithContextual && isViewMySitesNoContextual ) {
+			return 1; // "viewMySites" comes first over regular commands
+		}
+
+		return 0; // no change in order
+	};
+
+	// Filter commands to exclude "viewMySites" when the current path is /sites
+	const filteredCommands = commands.filter(
+		( command ) => ! ( command.name === 'viewMySites' && currentPath === '/sites' )
+	);
```

Comment:
@katinthehatsite , What do you think about replacing this logic with an imperative `if`.

After the commands are sorted...